packages/core-typings/src/IRoom.ts

```diff
@@ -77,11 +77,18 @@ export interface IRoom extends IRocketChatRecord {
 	description?: string;
 	createdOTR?: boolean;
 	e2eKeyId?: string;
+
+	/* @deprecated */
 	federated?: boolean;
 
 	channel?: { _id: string };
 }
 
+export interface IRoomFederated extends IRoom {
+	federated: true;
+}
+
+export const isIRoomFederated = (room: Partial<IRoom>): room is IRoomFederated => 'federated' in room && (room as any).federated === true;
```

Comment:
I suggest not referring the type by name here: `isRoomFederated` instead of `isIRoomFederated`. The reason is that the inner representation can change, outdating the type guard's name.

Also, keep in mind you're promoting a *incomplete* `IRoom` (`Partial<IRoom>`) to a complete `IRoomFederation` (`IRoom`). It may leave terrible unsoundness:

```
if (isRoomFederated(room)) {
  console.log(room._id); // string type on compile-time, but undefined on runtime
}
```